### PR TITLE
feat(dlp): Add code samples for REP endpoints

### PR DESCRIPTION
**Pull request by jakubrauch**
## Description

Fixes: N/A

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [x] I have followed guidelines from [CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CONTRIBUTING.md) and [Samples Style Guide](https://googlecloudplatform.github.io/samples-style-guide/)
- [x] **Tests** pass:   `npm test` (see [Testing](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CONTRIBUTING.md#run-the-tests-for-a-single-sample))
- [x] **Lint** pass:   `npm run lint` (see [Style](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CONTRIBUTING.md#style))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This pull request is from a branch created directly off of `GoogleCloudPlatform/nodejs-docs-samples`. Not a fork.
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new sample directory, and I created [GitHub Actions workflow](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CONTRIBUTING.md#adding-new-samples) for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [x] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: dlp/inspectStringRep.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+  }
+  inspectString();
+  // [END dlp_inspect_string]
+}
+
+main(...process.argv.slice(2));
+process.on('unhandledRejection', err => {
+  console.error(err.message);
+  process.exitCode = 1;
+});
```
Comment (subfuzion) [inline dlp/inspectStringRep.js:111]: Acknowledging that some inconsistency currently is present in our existing samples, nevertheless please remove the process handler.

It doesn't add value to the sample; an unhandled rejection will terminate the process and set the exitCode and provide a full stack trace, which is all this is doing (minus the stack trace).

If it's important to handle, you can simultaneously also address the warning that the call to main on line 53 ignores the returned promise by using [Promise.prototype.catch()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/catch)). For example:

```javascript
main(...process.argv.slice(2)).catch(err => {
  console.error(err);
  process.exitCode = 1;
});
```

**Review by subfuzion — CHANGES_REQUESTED** [review]
Looks good. Please address the one change I requested and I'll do one more pass before merging. Thanks.

File: dlp/inspectStringRep.js (hunk 1 above)
Comment (jakubrauch) [reply dlp/inspectStringRep.js]: Removed

File: dlp/inspectStringRep.js [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+      });
+    } else {
+      console.log('No findings.');
+    }
+  }
+  inspectString();
+  // [END dlp_inspect_string_rep]
+}
+
+main(...process.argv.slice(2));
```
Comment (subfuzion) [inline dlp/inspectStringRep.js:107]: Without awaiting, there is an unhandled promise warning; the best thing to do is catch an error here. You got rid of the unnecessary unhandled rejection handler (as requested), but you can address the warning here while doing the work that was getting done previously AND provide a full stack trace:

```javascript
main(...process.argv.slice(2)).catch(err => {
  console.error(err);
  process.exitCode = 1;
});
```

**Review by subfuzion — CHANGES_REQUESTED** [review]
Because of recent changes to address security concerns, you can't run this from a fork (see the error logs). Unfortunately, you'll need to resubmit this as a branch pull request directly from a clone of this repo, not a fork. If you're team membership doesn't give you enough permissions, let me know (feel free to ping directly).

**Comment by subfuzion** [comment]
@jakubrauch I made a few updates and got your commit working on a local branch. Closing this for #3788.

**Comment by subfuzion** [comment]
@jakubrauch Please note that in the interests of helping to clear the backlog and get the other fixes associated with the sample merged, I had to mark two flaky tests (possibly due to model changes) to be skipped. b/359246313

Please address https://github.com/GoogleCloudPlatform/nodejs-docs-samples/pull/3788/commits/4fcb66ab8a270f85038996bb22e885db3818440e in a follow-up PR. Thanks!

**Pull request closed without merging**